### PR TITLE
Implement card adding

### PR DESCRIPTION
**Pull request by Blejn**
- Show NewCard component when clicking on + button 
- It is possible to Cancel adding process
- It is possible to proceed Next to the second side
- It is possible to go Back when being on the second side
- It is possible to Save new card

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task goal
Implement adding new flashcards

### Requirements
1. It should show `NewCard` component when clicking on `+` button
2. It should be possible to `Cancel` adding process
3. It should be possible to proceed `Next` to the second side
4. It should be possible to go `Back` when being on the second side
5. It should be possible to `Save` new card

### Code Changes Review

#### package.json
The addition of "uuid" package is appropriate as it is used in the `NewCard.tsx` file for generating unique ids for the new cards.

#### src/App.tsx
The `addNewCard` function is correctly implemented and passed as a prop to the `NewCard` component. This function will add a new card to the existing list of cards.

#### src/components/NewCard.module.css
The change in this file is minor and doesn't affect the task requirements.

#### src/components/NewCard.tsx
The changes in this file cover all the requirements of the task:
- The `NewCard` component now accepts an additional prop `addNewCard` which is a function used to add a new card.
- A new interface `CardI` is defined to type check the new card object.
- The `saveCardClick` function is implemented which creates a new card with a unique id, question, and answer, and then calls the `addNewCard` function passed as a prop to add this new card.
- The `Save` button now has an onClick event handler which calls the `saveCardClick` function when clicked.

In conclusion, all the code changes are in line with the task requirements and there are no apparent bugs in the changed code.
